soh/soh/Enhancements/game-interactor/GameInteractor.h

```diff
@@ -162,11 +162,22 @@ class GameInteractor {
     static GameInteractionEffectQueryResult RemoveEffect(RemovableGameInteractionEffect* effect);
 
     // Game Hooks
-    template <typename H> struct RegisteredGameHooks { inline static std::vector<typename H::fn> functions; };
-    template <typename H> void RegisterGameHook(typename H::fn h) { RegisteredGameHooks<H>::functions.push_back(h); }
+    uint32_t nextHookId = 1;
+    template <typename H> struct RegisteredGameHooks { inline static std::unordered_map<uint32_t, typename H::fn> functions; };
+    template <typename H> uint32_t RegisterGameHook(typename H::fn h) {
+        // Ensure hook id is unique
+        while (RegisteredGameHooks<H>::functions.find(this->nextHookId) != RegisteredGameHooks<H>::functions.end()) {
```

Comment:
With `nextHookId` being a uint and you are treating an ID of 0 as being "unregistered", you'd probably want to exclude 0 from being assigned in this map. This would specifically occur once `nextHookId` overflows back to 0. (In a hypothetical future where everything efficiently registers/unregisters and some has a long running play sessions)

Similarly we may want a size check to prevent registering beyond `UINT32_MAX_VALUE - 1` hooks. Not sure what the failure code path should look like, but figure it would be good to account for that even if it is unlikely to ever happen.